x-pack/plugins/ml/public/components/full_time_range_selector/full_time_range_selector_service.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import moment from 'moment';
+
+import { i18n } from '@kbn/i18n';
+import { Query } from 'ui/embeddable';
+import { IndexPattern } from 'ui/index_patterns';
+import { toastNotifications } from 'ui/notify';
+import { timefilter } from 'ui/timefilter';
+import { ml } from '../../services/ml_api_service';
+
+export function setFullTimeRange(indexPattern: IndexPattern, query: Query) {
+  return ml
```

Comment:
Do we need the `return` here?

Reply:
No the `return` isn't needed. I'll remove it.

Reply:
The return was added in the original version so the promise chain could be used to trigger an update after the time range had been loaded.
I think this is potentially useful feature and there is no harm having the promise chain returned here.

Reply:
Well spotted @jgowdyelastic . The `return` is needed when prepopulating the job wizard settings when cloning a job that was created in the wizards. I have added it back in.